okhttp/src/main/kotlin/com/connectrpc/okhttp/ConnectOkHttpClient.kt

```diff
@@ -66,9 +67,13 @@ class ConnectOkHttpClient @JvmOverloads constructor(
                     // transform the error in a network interceptor to prevent okhttp from
                     // retrying it.
                     if (resp.code == 408 && isConnectUnary(chain.request())) {
-                        val contentType = resp.headers["Content-Type"]
-                        if (contentType == "application/json" || contentType == "application/json; charset=utf-8") {
-                            return resp.newBuilder().code(499).build()
+                        val mediaType = resp.headers["Content-Type"]?.toMediaTypeOrNull()
+                        if (mediaType != null && mediaType.toString() == "application/json") {
```

Comment:
`toString()` includes parameters, so you'd want to do:
```suggestion
                        if (mediaType != null && mediaType.type == "application" && mediaType.subtype == "json") {
```

Reply:
Ah, right, my bad. Thanks for spotting that!